client/post-editor/editor-term-selector/index.jsx

```diff
@@ -26,21 +30,34 @@ class EditorTermSelector extends Component {
 	}
 
 	onTermsChange( selectedTerm ) {
-		const { postTerms, taxonomyName } = this.props;
+		const { postTerms, taxonomyName, siteId, postId } = this.props;
 		const terms = cloneDeep( postTerms ) || {};
-		const taxonomyTerms = terms[ taxonomyName ] || {};
+		let taxonomyTerms = map( terms[ taxonomyName ] );
+		const termIds = map( taxonomyTerms, 'ID' );
+
+		if ( includes( termIds, selectedTerm.ID ) ) {
+			taxonomyTerms = taxonomyTerms.reduce( ( prev, term ) => {
+				if ( term.ID === selectedTerm.ID ) {
+					return prev;
+				}
 
-		if ( taxonomyTerms[ selectedTerm.name ] ) {
-			delete taxonomyTerms[ selectedTerm.name ];
+				prev.push( term );
+				return prev;
+			}, [] );
 		} else {
-			taxonomyTerms[ selectedTerm.name ] = selectedTerm;
+			taxonomyTerms.push( selectedTerm );
 		}
-		terms[ taxonomyName ] = taxonomyTerms;
 
-		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-		PostActions.edit( {
-			terms: terms
-		} );
+		const taxonomyTermIds = map( taxonomyTerms, 'ID' );
+
+		this.props.editPost( {
+			terms: {
+				[ taxonomyName ]: taxonomyTerms
+			},
+			terms_by_id: {
+				[ taxonomyName ]: taxonomyTermIds.length ? taxonomyTermIds : null
+			}
+		}, siteId, postId );
```

Comment:
Maybe the return value of `getEditedPost` could be normalized such that we don't include the full `terms` object, only `terms_by_id`. This avoids syncing concerns of both `terms` vs. `terms_by_id` and `terms` vs. `state.terms.items` (you'd need to call those selectors to get richer information about terms).

To that point, if we can't expect that the full terms are made available by the time we've received an existing post to be edited, maybe `state.terms.items` should be enhanced to pull the term objects from `RECEIVE_POST`? Don't recall if the structure of the objects is the same.